python-package/lightgbm/dask.py

```diff
@@ -271,10 +329,9 @@ def _train(
         )
 
     # Some passed-in parameters can be removed:
-    #   * 'machines': constructed automatically from Dask worker list
     #   * 'num_machines': set automatically from Dask worker list
     #   * 'num_threads': overridden to match nthreads on each Dask process
-    for param_alias in _ConfigAliases.get('machines', 'num_machines', 'num_threads'):
+    for param_alias in _ConfigAliases.get('num_machines', 'num_threads'):
         params.pop(param_alias, None)
```

Comment:
sorry, I accepted this suggestion but I think now that we should only apply it to `num_machines`, not `num_threads`.

This results in a warning that users cannot suppress.

>   /opt/conda/lib/python3.8/site-packages/lightgbm/dask.py:338: UserWarning: Parameter n_jobs will be ignored.
    _log_warning(f"Parameter {param_alias} will be ignored.")

Caused by the fact that `n_jobs` is an alias of `num_threads`

* https://github.com/microsoft/LightGBM/blob/646267d2656cbd13d86e498e96c6db95a9bae5d8/python-package/lightgbm/dask.py#L489
* https://github.com/microsoft/LightGBM/blob/646267d2656cbd13d86e498e96c6db95a9bae5d8/python-package/lightgbm/sklearn.py#L516

I believe that every warning should be something that can be changed by user code changes. Otherwise, we're just adding noise to logs that might cause people to start filtering out ALL warnings.

Reply:
I think it is important to notify users about `num_threads` as well before implementing #3714. Silently ignore parameter is more serious problem compared to unfixable warning, I believe.

Reply:
I disagree in this specific case about the meaning of "ignore", since this is a parameter default and not something explicitly passed in. However, since `num_threads` isn't directly related to the purpose of this PR and since I don't want to delay this PR too long because I'd like to merge #3823 soon after it, I'll leave this warning in for now and propose another PR in a few days where we could discuss it further.